.ai_scripts/generate_type_file.sh

```
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# ? ####################################################################
+# ? These are just Logons scripts to help him with some AI context shit.
+# ? ####################################################################
+
+# Goto the script directory
+script_dir=$(dirname "$0")
+cd "$script_dir"
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)